xbmc/filesystem/SMBFile.cpp

```diff
@@ -151,6 +158,10 @@ void CSMB::Init()
 
     // setup our context
     m_context = smbc_new_context();
+
+    // restore HOME
+    setenv("HOME", truehome, 1);
```

Comment:
Maybe save "truehome" in a std::string.
From what I read, the c-string returned from getenv is not immutable (quite obviously).